libs/template/renderer.go

```diff
@@ -160,7 +166,8 @@ func (r *renderer) computeFile(relPathTemplate string) (file, error) {
 		return nil, fmt.Errorf("failed to compute file content for %s. %w", relPathTemplate, err)
 	}
 
-	// Execute relative path template to get materialized path for the file
+	// Execute relative path template after stripping the .tmpl extension to
+	// get destination path for the file
```

Comment:
Why can't we reuse the logic below? It seems like we don't want to copy any files any more since we treat everything referenced as a template, right?

Reply:
So the intent here is that the file paths are always treated as templates. The copy file semantics exist only for the file content. 

The reason for this is:
1. The copy file semantics exist for prevent accidently parsing unintended files as go text templates. For example otherwise a binary executable might be treated as a go text template if we encounter UTF-8 encoded `{{` characters while reading it.
2. This approach also allows the name of non template files (ie files without the `.tmpl` extension) to be configurable, even if the content is meant to be copied over direcrly.

Reply:
I suspect @andrewnester means this this:
```go
relPath, err := r.executeTemplate(relPathTemplate)
```
That's now duplicated in this file. If we render the `relPath` before the block above then we only need to do it once, regardless of the file content being treated as a template or not.

Reply:
done, deduped the execution call